### DISCUSSION TITLE
Sync Addon development here, in gramp-project, or in addons-source?

### DISCUSSION THREAD

**Discussion by DavidMStraub** [DavidMStraub/gramps-web-sync#36] [General]
The sync addon is an important component of Gramps Web for those users (like me :blush:) that use Web and desktop in parallel. At the moment, this repository is used for the development version of the sync addon as well as for issues & feature requests. Since for some future improvements, changes to the addon and Gramps Web API are needed, it would make sense to couple the development more tightly, e.g. using the new cross-repository issue project I've set up [here](https://github.com/orgs/gramps-project/projects/1). That would be possible if we move this repository inside the `gramps-project` organization.

We could even distribute the addon releases through a custom addon source URL and build it automatically like described [in this thread](https://gramps.discourse.group/t/github-workflow-to-automatically-build-addons/6499), so there would be no need to merge it into `addons-source` or `addons`.

What do you think? Especially @Nick-Hall?

**Comment by Nick-Hall** [comment]
Why don't you develop it in the `addons-source` repository with the other addons?  We could enable issue tracking on the repository.  This could be a first step in migrating all issues from Mantis BT to GitHub issues.

**Reply by dstraub-pf** [reply]
We can definitely do it, but PR reviews have been a bottleneck there - understandably, because there is so much else going on.

**Reply by Nick-Hall** [reply]
A PR for a change to an addon in the `addons-source` repository will require a review by the addon owner.  This may take some time depending on their availability.  In v5.2, we introduced the concept of maintainers.  An owner may delegate their review and sign-off responsibilities to one or more maintainers.  If an addon has been abandoned then we may also appoint a maintainer.

Before publishing an addon update in the `addons` repository, a trusted member of our team will review it as a precaution against malicious code and addons that update data without making the changes clear to the user.  It would be very rare for us to reject code at this stage.  The purpose of this step is to give our users confidence that using addons from the official repository is safe.

We aim to be quick, but we are all volunteers and have other commitments. This is partly why we have made it easier to host external addon repositories.

**Reply by dstraub-pf** [reply]
Ah cool, I somehow missed that point. So I can approve PRs for the Gramps Web Sync Addon? :slightly_smiling_face: 

By the way, I could bring that workflow feature [here](https://gramps.discourse.group/t/github-workflow-to-automatically-build-addons/6499) to the finish line and add a PR to `addons` (or `addons-source`) - in principle we could even get rid of `addons-source` and have the following structure:

`addons/master`: built addons - so no breaking change to people's addon managers
`addons/maintenance/gramps...` - sources

**Comment by Nick-Hall** [comment]
> So I can approve PRs for the Gramps Web Sync Addon?

An addon owner can approve a PR for their addon in the `addons-source` repository, but not in the `addons` repository.  In fact, I usually grant write access to the `addons-source` repository to any developer that requests it.

Publishing an addon to the `addons` repository requires an extra step.  A trusted member of our team will review the code before publishing it.

**Reply by DavidMStraub** [reply]
Ok, but then we're back to square one because users will not see the updated addon until it's reviewed to go from addons-source to addons. So we need to decide between:

1. only develop in `addons-source` and rely on timely integration in `addons`
2. develop in a separate repo with a build action
3. Do both 1. & 2. so users can opt in to using a version before it's released in `addons`

**Reply by Nick-Hall** [reply]
There are three levels of plugin:  built-in plugins, official addons and external addons.  Each has its advantages and disadvantages.

The plan is to eventually allow official addons to override built-in plugins and external addons to override official addons.  It didn't seem sensible to include this functionality in the first version of the Addon Manager.  The ability to host an external addons was also new in v5.2.

**Reply by DavidMStraub** [reply]
Hi,

so my suggestion is to enable issues in `addons-source` and I want to request write access to `addons-source` to merge Sync addon PRs. I plan to work on the sync addon in the next days.

**Reply by DavidMStraub** [reply]
I created a PR for an improved sync addon and cleaned up old issues, but there are still 7 relevant open issues. @Nick-Hall let me know once you've enabled issues on `addons-source`, then I'll move them over and archive the other repo.

**Reply by DavidMStraub** [reply]
(Issues: https://github.com/DavidMStraub/gramps-web-sync/issues)

**Reply by Nick-Hall** [reply]
I have posted to the _gramps-devel_ mailing list proposing the use of GitHub Issues.  See [Proposal to use GitHub Issues](https://sourceforge.net/p/gramps/mailman/message/59113680/).

**Reply by DavidMStraub** [reply]
Ah, I wasn't aware this was coupled between addons and Gramps core, thanks!

**Comment by DavidMStraub** [comment]
Hi, I'd like to revive this discussion.

The sync addon needs work, and Github issues is the right place to track it (as it can be integrated with Web API issues). As a consequence of the discussion here, addon development was moved to `addons-source`, and I have to say it works great (kudos to Gary for maintaining it). However, issues are still being tracked in this repo, which is no longer up to date. This is not a great situation and might confuse users.

It would be good to have a final decision yes/now, whether it's planned to enable issues on `addons-source` in the near future. If yes, we can move everything there. If not, we might have to make this an up-to-date mirror again.

**Comment by Nick-Hall** [comment]
> It would be good to have a final decision yes/now, whether it's planned to enable issues on addons-source in the near future.

There are no plans to move to GitHub Issues in the near future.
